data/pathfinder/paizo/roleplaying_game/core_rulebook/cr_feats.lst

```diff
@@ -166,7 +166,7 @@ Step Up											CATEGORY:FEAT		TYPE:Combat
+Stunning Fist										CATEGORY:FEAT		TYPE:Combat.AttackOption			PREABILITY:1,CATEGORY=FEAT,Improved Unarmed Strike																																																										PRESTAT:2,DEX=13,WIS=13	PRETOTALAB:8									DEFINE:StunningFistMonkLVL|0	DEFINE:StunningFistAttack|0	DEFINE:StunningFistDC|0											DESC:You know just where to strike to temporarily stun a foe. %1/day (DC %2)|StunningFistAttack|StunningFistDC																					STACK:NO																																																		ABILITY:Ability Focus|AUTOMATIC|Stunning Fist																																																																																																																																																				BONUS:VAR|StunningFistDC|2|PREABILITY:1,CATEGORY=FEAT,Ability Focus (Stunning Fist)	BONUS:VAR|StunningFistDC|10+(TL/2)+WIS	BONUS:VAR|StunningFistAttack|floor(MonkLVL+(MonkLVL/4))|TYPE=NoStack															SOURCEPAGE:p.135	BENEFIT:You must declare that you are using this feat before you make your attack roll (thus, a failed attack roll ruins the attempt). Stunning Fist forces a foe damaged by your unarmed attack to make a Fortitude saving throw (DC %1), in addition to dealing damage normally. A defender who fails this saving throw is stunned for 1 round (until just before your next turn). A stunned character drops everything held, can't take actions, loses any Dexterity bonus to AC, and takes a -2 penalty to AC. You may attempt a stunning attack once per day for every four levels you have attained (but see Special), and no more than once per round. Constructs, oozes, plants, undead, incorporeal creatures, and creatures immune to critical hits cannot be stunned.|StunningFistDC
```

Comment:
First, I agree that it's not complete. I got it working for monk but am guessing it's broken for non-monks.

When you take an archetype do you no longer get class levels? 

Stunning first is treated two ways, one way for monks and one way for everyone else as far as I understand it.  For Monks its Monk level plus 1/4 monk level rounded down. For everyone else its 1/4 total level. 

Assuming we agree this doesn't work
OLD: StunningFistMonkLVL+(TL-StunningFistMonkLVL)/4
Could we not just change this
NEW: floor(MonkLVL+(MonkLVL/4))
to
floor(MonkLVL+(TL/4))
That should give monk level bonus if you have monk levels and also give the 1/4 bonus everyone including monks get. Depending on how archetypes are treated. 

A quick look seems like only monks get any special circumstances with stunning fist. 

It could be the case that the feat entry could be floor(TL/4) and that something should be done to the class abilities for monk to fix it. 

If what I have said just repeats what you've said and I just need to work on understanding all that data you typed up just let me know. I know I have a lot to learn. 

I'll start working on implementing what you said now! 😄